core/pom.xml

```diff
@@ -527,9 +527,9 @@ THE SOFTWARE.
       <version>3.2.9</version>
     </dependency>
     <dependency>
-      <groupId>org.jenkins-ci</groupId>
+      <groupId>org.jmdns</groupId>
       <artifactId>jmdns</artifactId>
-      <version>3.4.0-jenkins-3</version>
+      <version>3.5.5</version>
```

Comment:
Can you please link to a diff between the currently used one and the update you're proposing?

Also, I think we should really rather first bump to the closest codebase, i.e. switch to say `org.jmdns:jmdns:3.4.0` or `org.jmdns:jmdns:3.4.1`, so we limit the difference (and again, a diff is IMO critical so we can assess the risk during this review).

Then, in some next release, we'd bump to 3.5.5.

Reply:
(NOTE: I'm aware this is the library update that is supposed to fix the issue, but I'm quite concerned getting back to the upstream version *and* updating may be a bit too risky in one go (?))

Reply:
I've added the diff to the Description of the PR. I see your point, but for example in the forked library there is an opened PR to update to 3.5.3: https://github.com/jenkinsci/jmdns/pull/2. If this PR was finished and merged, the update would be from 3.4.0 to 3.5.2. I'm just "continuing" that work but easing the way we can update.

Is there any other tool to test it apart from local testing and functional/unitary tests? I haven't found any :(

Reply:
If there's significant concern about the risk then pointing to 3.4.0 and upgrading in a separate release might be good. On the other hand, it's a known fact that 3.4.0 has a bug so it might not be worth it.